src/tools/cephfs/JournalTool.cc

```diff
@@ -487,14 +489,245 @@ int JournalTool::main_event(std::vector<const char*> &argv)
+    EUpdate *decoupled_eu;       // event containing decoupled directory mkdir
+    inodeno_t decoupled_ino = 0; // inode and dirfrag of decoupled directory
+    uint64_t max = 0;            // where to start appending to the journal
+    dout(10) << "Writing out some bogus files" << dendl;
+
+    // poach the dirlumps from the snapshot AND find the end of the journal
+    r = js.scan();
+    for (JournalScanner::EventMap::const_iterator i = js.events.begin(); i != js.events.end(); ++i) {
+      if (i->first > max)
+        max = i->first;
+      // TODO: we need to prune dirlumps that we don't care about (e.g., if there hierarchy > 2)
+      // TODO: if there are multiple mkdirs
+      if (i->second.log_event->get_type() == EVENT_UPDATE) {
+        EUpdate *eu = reinterpret_cast<EUpdate*>(i->second.log_event);
+        dout(0) << "found update type=" << eu->type << " while searching for decoupled_dir=" << decoupled_dir << dendl;
+        if (eu->type == "mkdir") {
+          // get the inode for the decoupled dir (this gives us the parent df)
+          map<dirfrag_t, EMetaBlob::dirlump> lumps = eu->metablob.get_lump_map();
+          for(map<dirfrag_t, EMetaBlob::dirlump>::iterator k = lumps.begin();
+              k != lumps.end();
+              k++) {
+            std::string format = "json-pretty";
+            Formatter *f = Formatter::create(format);
+            k->second.dump(f);
+            bufferlist out;
+            f->flush(out);
+            dout(10) << "checking df=" << k->first << " dirlump=" << out.to_str() << dendl;
+            list<ceph::shared_ptr<EMetaBlob::fullbit> > dfull = k->second.get_dfull(); 
+            for(list<ceph::shared_ptr<EMetaBlob::fullbit> >::iterator j = dfull.begin();
+                j != dfull.end();
+                j++) {
+              // TODO: we only support two level trees -- need to match all path components
+              dout(10) << "  fullbit->dn=" << (*j)->dn << dendl;
+              if ((*j)->dn == decoupled_dir) {
+                decoupled_ino = (*j)->inode.ino;
+                decoupled_eu  = reinterpret_cast<EUpdate*>(i->second.log_event);
+                dout(4) << "found decoupled directory dirlump at ino=" << decoupled_ino << dendl;
+              }
+            }
+          }
+	  // TODO: this might inefficient with a lot of direntries and it
+	  // assumes the metablob has everything (e.g., all dfs)
+        }
+      }
+    }
+
+    if (!decoupled_eu || decoupled_ino == 0) {
+      derr << "ERROR: couldn't find decoupled dir=" << decoupled_dir
+           << "... maybe a corrupt log import?" << dendl;
+      return -ENOENT;
+    }
+
+    // setting update interval
+    double update_interval = 2;
+    // record start time and check for update intervals at the end of the for loop for every file created
+    std::clock_t start;
+    std::clock_t end;
+    start = std::clock();
+    //spawn child process for background update
+    pid_t pid = fork();
+    if(pid == 0){
```

Comment:
However, as you said, all your changes also seem to have come into my commit because your Ceph branch didn't have any of the Cudele stuff. Please tell me if there is a better way to do this to fix this issue.